service/matching/matchingEngine.go

```diff
@@ -211,6 +216,24 @@ func (e *matchingEngineImpl) getTaskListManager(taskList *taskListID,
 	return mgr, nil
 }
 
+func (e *matchingEngineImpl) cacheTaskListByDomain(taskList *taskListID) error {
+	domainName, err := e.domainCache.GetDomainName(taskList.domainID)
+	if err != nil {
+		return err
+	}
+	cachedTL := e.taskListCache.Get(domainName)
+	var taskLists map[string]bool
+	if cachedTL != nil {
+		taskLists = cachedTL.(map[string]bool)
```

Comment:
In this case, PutIfNotExist below won't do anything as the domainName already exists in the cache?